src/DSE.Open.Globalization/LanguageTag.cs

```diff
@@ -122,10 +125,190 @@ public bool LanguagePartEquals(LanguageTag otherLangPart)
+               letter subtags are titlecase (as in the tag "az-Latn-x-latn").
+             */
+
+            var valueSpan = value.AsSpan();
+
+            var ti0 = valueSpan.IndexOf((AsciiChar)'-');
+
+            // only language tag - all lowercase and return...
+
+            if (ti0 < 0)
+            {
+                for (var i = 0; i < value.Length; i++)
+                {
+                    destination[i] = (char)value[i].ToLower();
+                }
+
+                return;
+            }
+
+            // language part to lower
+
+            for (var i = 0; i < ti0; i++)
+            {
+                destination[i] = (char)value[i].ToLower();
+            }
+
+            // from now on all subtags to lower except for 2 and 4 letter subtags,
+            // unless occuring after singletons            
+
+            var remaining = valueSpan[(ti0 + 1)..];
+            var previousSingleton = false;
+            var destIndex = ti0;
+
+            destination[ti0] = '-';
+            destIndex++;
+
+            int length;
+            var lastTag = false;
+
+            while (!remaining.IsEmpty)
+            {
+                length = remaining.IndexOf((AsciiChar)'-');
+
+                if (length < 0)
+                {
+                    length = remaining.Length;
+                    lastTag = true;
+                }
+
+                if (!previousSingleton && length is 2)
+                {
+                    for (var j = 0; j < length; j++)
+                    {
+                        destination[destIndex + j] = (char)remaining[j].ToUpper();
+                    }
+                }
+                else if (!previousSingleton && length is 4)
+                {
+                    destination[destIndex] = (char)remaining[0].ToUpper();
+
+                    for (var j = 1; j < length; j++)
+                    {
+                        destination[destIndex + j] = (char)remaining[j].ToLower();
+                    }
+                }
+                else
+                {
+                    for (var j = 0; j < length; j++)
+                    {
+                        destination[destIndex + j] = (char)remaining[j].ToLower();
+                    }
+                }
+
+                destIndex += length;
+
+                if (!lastTag)
+                {
+                    destination[destIndex] = '-';
+                    destIndex++;
+                    remaining = remaining[(length + 1)..];
+                    previousSingleton = length == 1;
+                }
+                else
+                {
+                    Debug.Assert(length == remaining.Length);
+                    remaining = default;
+                }
+            }
+        }
+
+        static void FormatLower(AsciiString value, Span<char> destination)
+        {
+            for (var i = 0; i < value.Length; i++)
+            {
+                destination[i] = value[i].ToUpper();
```

Comment:
@frankbuckley

```suggestion
                destination[i] = value[i].ToLower();
```

Reply:
Oops. And, actually, something like ToLower/ToUpper should sit on AsciiString, but this will do for now.